app/models/bookmark.rb

```diff
@@ -46,7 +46,14 @@ def self.auto_delete_preferences
   validates :name, length: { maximum: 100 }
 
   def unique_per_post_for_user
-    if Bookmark.exists?(user_id: user_id, post_id: post_id)
+    is_first_post = Post.where(id: post_id).pluck_first(:post_number) == 1
```

Comment:
Minor but you don't need to pluck the column here. 

`is_first_post = Post.exists?: post_id, post_number: 1)`